### PR TITLE
feat(ts): implements testing and release (#184)

### PR DESCRIPTION
**Pull request by ygrishajev**
This PR introduces several updates aimed at improving the automated publishing and validation processes of our TypeScript package. The enhancements include the addition of new make targets, workflow updates, and configuration adjustments to streamline operations both locally and in our CI system. Here's a detailed breakdown of the changes:

- **Makefile Updates**:
  - Added new targets for linting, testing, and releasing the package.
  - These targets can be executed locally via `make`. The same targets are also utilized in our CI pipeline to maintain consistency and reliability across environments.

- **CI/CD Workflow Enhancements**:
  - Updated existing workflows and created new ones to run validations and manage the automated release processes efficiently.

- **Semantic Release Configuration**:
  - Configured semantic release to automate the versioning and package publishing process, leveraging commit messages to determine version changes.

- **NPM Package Configuration**:
  - Tailored the npm package contents to ensure only necessary files and metadata are included, optimizing the package for distribution.

- **Test Coverage**:
  - Set up test coverage reporting and ensured that coverage metrics are captured and uploaded to Codecov during the CI process.

- **Code Refactoring**:
  - Refactored the CoinDec patch to reduce code duplication, enhancing code maintainability and reducing the potential for errors.

These enhancements are designed to ensure our development processes are robust, consistent, and efficient, facilitating easier and more reliable package publishing and validation.

### CODE REVIEW

File: ts/package.json [hunk 1]
```diff
@@ -42,7 +51,7 @@
     "format": "prettier --write './**/*.{js,jsx,ts,tsx,css,json}' --config ./.prettierrc",
     "lint": "eslint src",
     "lint:fix": "npm run lint -- --fix",
-    "postbuild": "script/generate-exports.ts",
+    "postbuild": "script/generate-exports.js",
```
Comment (baktun14) [inline ts/package.json:54]: ts wasn't working?

File: ts/.releaserc [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+      }
+    ],
+    [
+      "@semantic-release/github",
+      {
+        "assets": "package/*.tgz"
+      }
+    ]
+  ],
+  "preset": "angular",
```
Comment (baktun14) [inline ts/.releaserc:24]: angular?

**Review by baktun14 — COMMENTED** [review]
Can you add a description to the PR please?

File: ts/package.json (hunk 1 above)
Comment (ygrishajev) [reply ts/package.json:54]: in CI no. I would need to compile or add ts globally. not a big deal but also it has no types so decided to just make it js

File: ts/.releaserc (hunk 2 above)
Comment (ygrishajev) [reply ts/.releaserc:24]: https://github.com/angular/angular/blob/68a6a07/CONTRIBUTING.md#commit

File: ts/package.json (hunk 1 above)
Comment (baktun14) [reply ts/package.json:54]: Ok yea makes sense

**Comment by ygrishajev** [comment]
> Can you add a description to the PR please?

@baktun14 done

File: ts/.releaserc (hunk 2 above)
Comment (baktun14) [reply ts/.releaserc:24]: Ah I see, clever marketing stunt.

**Review by baktun14 — APPROVED** [review]

**Review by baktun14 — DISMISSED** [review]

File: make/release-ts.mk [hunk 3]
```diff
@@ -0,0 +1,11 @@
+.PHONY: release-ts
+
```
Comment (troian) [inline make/release-ts.mk:2]: ✂️

Comment (ygrishajev) [reply make/release-ts.mk]: you mean to cut the line? done if so

**Comment by troian** [comment]
there were proto changes merged yesterday, need to rebase on latest main and run codegen

**Comment by ygrishajev** [comment]
> there were proto changes merged yesterday, need to rebase on latest main and run codegen

@troian done

**Pull request merged**